tests/e2e/framework/main_entry.go

```diff
@@ -33,9 +37,71 @@ func MainEntry(m *testing.M) {
 
 	Global = f
 
+	// Do suite setup
+	if err := f.setUp(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Run the tests
 	exitCode, err := f.runM(m)
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	// Do suite teardown only if we have a successful test run or if we don't care
+	// about removing the test resources if the test failed.
+	if exitCode == 0 || (exitCode > 0 && !f.skipCleanupOnError) {
+		if err = f.tearDown(); err != nil {
+			log.Fatal(err)
```

Comment:
Note for reviewers.

This call to `log.Fatal()` is new from a testing behavior perspective.

In the past, we never cared if we didn't cleanup resources properly at the end of the test run. This changes that behavior by causing the tests to fail if we don't cleanup properly.